source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -30,6 +35,115 @@ def __init__(self, obj, position, _rangeObj=None):
 					1
 				)
 
+	def move(self, unit, direction, endPoint=None):
+		if not self._isCaret:
+			# Insure we haven't gone beyond the visible text.
+			# UIA adds thousands of blank lines to the end of the console.
+			visiRanges = self.obj.UIATextPattern.GetVisibleRanges()
+			lastVisiRange = visiRanges.GetElement(visiRanges.length - 1)
+			if self._rangeObj.CompareEndPoints(
+				UIAHandler.TextPatternRangeEndpoint_Start,
+				lastVisiRange,
+				UIAHandler.TextPatternRangeEndpoint_End
+			) >= 0:
+				return 0
+		if unit == textInfos.UNIT_WORD and direction != 0:
+			# UIA doesn't implement word movement, so we need to do it manually.
+			offset = self._getCurrentOffset()
+			index = 1 if direction > 0 else 0
+			start, end = self._getWordOffsets(offset)
+			wordMoveDirections = (
+				(offset - start) * -1,
+				end - offset
+			)
+			res = self.move(
+				textInfos.UNIT_CHARACTER,
+				wordMoveDirections[index],
+				endPoint=endPoint
+			)
+			if res != 0:
+				return direction
+			else:
+				if self.move(textInfos.UNIT_CHARACTER, -1):
+					return self.move(unit, direction, endPoint=endPoint)
+				else:
+					return res
+		return super(consoleUIATextInfo, self).move(unit, direction, endPoint)
+
+	def expand(self, unit):
+		if unit == textInfos.UNIT_WORD:
```

Comment:
Expand should use similar code to move. E.g.:
If unit is word, then use _getCurrentOffset and _getWordOffsets and manually move the start and end of the range to have it correctly bound the current word.
Then, there is no need to change what text is exposed as the textRange will be correct.
The advantage of this is there is less to track, it is easier to read, and most importantly, later comparisons or movements of this textInfo won't behave like it is expanded to line even though the text returned is word.
There may be a question around performance: but I'd like to see the code changed to my proposed way before we then consider if optimisations are needed at all.